src/backend/out.c

```diff
@@ -338,89 +340,96 @@ void outdata(symbol *s)
 
     offset = s->Soffset;
 
-    for (dt_t *dt = dtstart; dt; dt = dt->DTnext)
+    dt_writeToObj(*objmod, dtstart, seg, offset);
+    Offset(seg) = offset;
+#if SCPP
+    out_extdef(s);
+#endif
+Lret:
+    dt_free(dtstart);
+}
+
+/********************************************
+ * Write dt to Object file.
+ * Params:
+ *      objmod = reference to object file
+ *      dt = data to write
+ *      seg = segment to write it to
+ *      offset = starting offset in segment - will get updated to reflect ending offset
+ */
+
+void dt_writeToObj(Obj& objmod, dt_t *dt, int seg, targ_size_t& offset)
```

Comment:
Yes. And this code will eventually get ported to D, and using `ref` is preferred in D rather than `*` because `ref` implies that the callee does not save a copy of the pointer.